lib/Optimizer/IROptimizer.cpp

```diff
@@ -347,18 +347,14 @@ static void deleteDeadAllocs(IRFunction &M) {
   llvm::SmallVector<Instruction *, 16> erasedInstructions{};
 
   // Remove all unused tensorviews.
-  for (auto &I : instrs) {
-    if (isa<TensorViewInst>(I) && I.getNumUsers() == 0) {
-      erasedInstructions.push_back(&I);
+  for (auto it = instrs.rbegin(); it != instrs.rend();) {
+    if (isa<TensorViewInst>(*it) && it->getNumUsers() == 0) {
```

Comment:
I think this if-then-else-statement could be simplified. May be something like this:

```c++
for (auto it = instrs.rbegin(); it != instrs.rend();) {
  // Remember the current instruction and advance the iterator.
  auto *I = &*it++;
  // Erase unused tensor views.
  if (isa<TensorViewInst>(I) && I->getNumUsers() == 0) {
    // Remove a tensor view. 
    // It may make other tensor views preceding it eligible for a removal as well.
    M.eraseInstruction(I);
  }
}
```

Reply:
You could also provide a small comment explaining why we iterate in a reverse order.

Reply:
+1.

Also, now we can remove the worklist (erasedInstructions).